backend/qemu.pm

```diff
@@ -346,6 +347,63 @@ sub deflate_balloon ($self) {
     $self->handle_qmp_command({execute => 'balloon', arguments => {value => $ram_bytes}}, fatal => 1);
 }
 
+sub backup_disks ($self, $args) {
+    my $vars = \%bmwqemu::vars;
+    my $bdc = $self->{proc}->blockdev_conf;
+    my $fname = $args->{filename};
+    my $rsp = $self->handle_qmp_command({execute => 'query-status'}, fatal => 1);
+    bmwqemu::diag("Saving disks (Current VM state is $rsp->{return}->{status})");
+    my $was_running = $rsp->{return}->{status} eq 'running';
+
+    if ($was_running) {
+        $self->inflate_balloon();
+        $self->freeze_vm();
+    }
+
+    mkpath("ulogs");
+    $bdc->for_each_drive(sub {
+            local $Data::Dumper::Indent = 0;
+            local $Data::Dumper::Terse = 1;
+            local $Data::Dumper::Sortkeys = 1;
+            my $drive = shift;
+            my $size = $drive->{drive}->{size};
+            my $id = $drive->{id} . "-backup-$fname";
+            my $node = $drive->{drive}->{node_name};
+            my $my_node = $node . "-$fname";
+            # create disk
+            runcmd("qemu-img", "create", "-f", "qcow2", "ulogs/$my_node.qcow2", "$size");
+            my $req = {execute => 'blockdev-add',
+                arguments => {driver => 'qcow2', 'node-name' => $my_node,
+                    file => {driver => 'file', filename => "ulogs/$my_node.qcow2"}
+                }};
+            my $response = $self->handle_qmp_command($req, fatal => 1);
+
+            $req = {execute => 'blockdev-backup',
+                arguments => {device => $node, target => $my_node,
+                    sync => 'full', 'job-id' => $id}};
+
+            $response = $self->handle_qmp_command($req, fatal => 1);
+
+            my $return;
+            do {
+                my $query = {execute => 'query-jobs'};
+                ($return) = $self->handle_qmp_command($query, fatal => 1)->{return};
+                sleep 10;
```

Comment:
I assume this is waiting while the saving to persistent storage is going on. I think we should spend a little bit more CPU cycles to prevent needless waiting

```suggestion
                sleep 1;
```

Reply:
I suggest we die on a timeout here, maybe a variable like QUERY_DISK_MAX or similar in case we realize it's slow on some machines. And "die" so we get a reason instead of a possibly confusing generic timeout.

Reply:
I'm also in favor of a lower sleep time and a timeout for this otherwise potentially endless loop.

Reply:
@kalikiana  @Martchus  any idea how long timeout will be good ?  .. in this loop we waiting for backup of one disk from n going as background job in QEMu ( as this is only supported backend , ) +- 5-20GB of one disk